tools/runners/run-proxy.js

```diff
@@ -211,9 +213,19 @@ Object.assign(Proxy.prototype, {
       if (self.mode === "errorpage") {
         showErrorPage(c.res);
       } else {
-        attempt(c.res, () => self.proxy.web(c.req, c.res, {
-          target: 'http://' + self.proxyToHost + ':' + self.proxyToPort
-        }));
+        attempt(c.res, () => {
+          if (self.proxyToSocket) {
+            return self.proxy.web(c.req, c.res, {
```

Comment:
We could make these a bit nicer by constructing `options` object beforehand (and only branching there), and using it both here for the `web` path, and below for the `ws` path.